### PR TITLE
Update to DD4hep v1

### PR DESCRIPTION
**Pull request by vvolkl**
Note that builds are expected to fail until dd4hep v1 is used.

Changes proposed in this pull-request:

- This PR propagates namespace-changes and other updates in DD4hep v1 to FCCSW.

To be done:
  [ ] install and use DD4hep v1

- [ ]
- [ ]

### CODE REVIEW

File: Detector/DetFCChhECalSimple/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ find_package(ROOT COMPONENTS MathCore GenVector Geom REQUIRED)
 
 install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/compact DESTINATION Detector/DetFCChhECalSimple)
 
-include_directories(SYSTEM ${DD4hep_INCLUDE_DIRS} ${ROOT_INCLUDE_DIRS})
+#include_directories(SYSTEM ${DD4hep_INCLUDE_DIRS} ${ROOT_INCLUDE_DIRS})
```
Comment (jhrdinka) [inline Detector/DetFCChhECalSimple/CMakeLists.txt]: Why here DD4hep and root do not need to be included anymore?

Comment (vvolkl) [reply Detector/DetFCChhECalSimple/CMakeLists.txt]: They are included at l. 21, this was a workaround for a bug that should be fixed now.

**Comment by vvolkl** [comment]
Todo list
[ ] Fix `libvecgeom.a` (possibly with new installation of geant)
[ ] Fix missing `ExRootAnalysis/ExRootTreeBranch.h` in DelphesInterface
[ ] Fix warnings about missing `ROOT::Geom
[x] Fix retrieval of event number in SimPapas, that changed with new Gaudi version
[ ] Fix ```
/cvmfs/fcc.cern.ch/testing/sw/views/Wed/x86_64-slc6-gcc62-opt/lib/libpythia8.a(PartonDistributions.o): In function `Pythia8::LHAPDF::symbol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
PartonDistributions.cc:(.text+0x10541): undefined reference to `dlerror'
```
[x] Fix changed `const` members in ACTS
[ ] Fix merge conflicts, pdate with changes to recently merged PRs

**Comment by vvolkl** [comment]
@phsft-bot build

**Comment by vvolkl** [comment]
@phsft-bot build

**Comment by vvolkl** [comment]
@phsft-bot build

**Comment by vvolkl** [comment]
@phsft-bot build

**Comment by vvolkl** [comment]
@phsft-bot build

**Comment by JavierCVilla** [comment]
@phsft-bot build

**Comment by vvolkl** [comment]
Superseded by #278

**Pull request closed without merging**